### PR TITLE
Split the backend into a docker and a swarm backend

### PR DESCRIPTION
**Pull request by akerouanton**
This is not compiling, so the refactoring is still in a WIP state but the diff got a little big already due to tests.

### CODE REVIEW

File: backend/docker/backend.go [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+	for _, exporter := range exporters {
+		err := b.stopExporter(ctx, exporter, force)
+		if err != nil && !IsErrExportedTaskStillRunning(err) {
+			return err
+		}
+		if err != nil && IsErrExportedTaskStillRunning(err) {
+			failed = append(failed, exporter.Names[0])
+		}
+	}
+
```
Comment (akerouanton) [inline backend/docker/backend.go]: A message specifying how much exporters cleanup failed/succeeded should be logged.

File: cmd/autoexport.go [hunk 2]
```diff
@@ -50,3 +45,64 @@ func AutoExport(c *cli.Context) {
...
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if _, ok := c.funcs[k]; ok {
+		delete(c.funcs, k)
+	}
+}
+
+// @TODO: implement true back-off retry
+func retry(times uint, interval time.Duration, f func() error) error {
```
Comment (nicolasmure) [inline cmd/autoexport.go]: is this fn still used ?

File: backend/docker/event_handler.go [hunk 3]
```diff
@@ -19,67 +19,20 @@ import (
...
-	defer c.mutex.Unlock()
-
-	if _, ok := c.funcs[k]; ok {
-		delete(c.funcs, k)
-	}
-}
-
-func (b Backend) ListenEventsForExported(ctx context.Context, promNetwork string) {
-	evtCh, errCh := b.cli.Events(ctx, types.EventsOptions{
+func (b DockerBackend) ListenForTasksToExport(ctx context.Context) {
```
Comment (nicolasmure) [inline backend/docker/event_handler.go]: In the end this file would be replaced by a swarm event handler ?

File: cmd/autoexport.go (hunk 2 above)
Comment (akerouanton) [reply cmd/autoexport.go]: Yep it's gonna be use to implement a sort of retry to start/stop exporters ;)

File: backend/docker/event_handler.go (hunk 3 above)
Comment (akerouanton) [reply backend/docker/event_handler.go]: Not really, cause for Swarm, the `autoexporter` is gonna run with the docker backend and the `autoconfig` is gonna run with the swarm backend ;)

**Pull request closed without merging**